lighthouse-treemap/app/src/main.js

```diff
@@ -55,19 +55,25 @@ class TreemapViewer {
     this.el = el;
     this.getHueForKey = TreemapUtil.stableHasher(TreemapUtil.COLOR_HUES);
 
-    // TODO: make "DataSelector" to switch between different groups or specific d1 nodes
-    const group = 'scripts';
-    const depthOneNodes = this.depthOneNodesByGroup[group];
-    this.currentTreemapRoot = this.wrapNodesInNewRootNode(depthOneNodes);
-    TreemapUtil.walk(this.currentTreemapRoot, (node, path) => this.nodeToPathMap.set(node, path));
-
-    this.viewModes = this.createViewModes();
-    this.currentViewMode = this.viewModes[0];
+    /* eslint-disable no-unused-expressions */
+    /** @type {LH.Treemap.Node} */
+    this.currentTreemapRoot;
+    /** @type {LH.Treemap.ViewMode} */
+    this.currentViewMode;
+    /** @type {LH.Treemap.Selector} */
+    this.selector;
+    /** @type {LH.Treemap.ViewMode[]} */
+    this.viewModes;
+    /** @type {RenderState=} */
+    this.previousRenderState;
```

Comment:
cant this also be initialized with the same shape? it'd mean you can remove the awkward `!this.previousRenderState ||` bits below, right?

Reply:
I did try, but it made things more awkward.